src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitString.Fixer.cs

```diff
@@ -0,0 +1,131 @@
+
+using System.Composition;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Operations;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis.Editing;
+using System.Linq;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, LanguageNames.VisualBasic), Shared]
+    public class PreferConstCharOverConstUnitStringFixer : CodeFixProvider
+    {
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(PreferConstCharOverConstUnitStringAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken cancellationToken = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            if (root.FindNode(context.Span) is SyntaxNode expression)
+            {
+                SemanticModel semanticModel = await doc.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var operation = semanticModel.GetOperation(expression, cancellationToken);
+                if (operation is IArgumentOperation argumentOperation)
+                {
+                    IOperation? isLocalReferenceOperation = argumentOperation.Value as ILocalReferenceOperation;
+                    IOperation? isLiteralOperation = argumentOperation.Value as ILiteralOperation;
+                    if (isLocalReferenceOperation == null && isLiteralOperation == null)
+                    {
+                        return;
+                    }
+
+                    IVariableDeclaratorOperation? cachedVariableDeclaratorOperation = default;
+                    if (isLocalReferenceOperation != null)
+                    {
+                        ILocalReferenceOperation localReferenceOperation = (ILocalReferenceOperation)argumentOperation.Value;
+                        ILocalSymbol localArgumentDeclaration = localReferenceOperation.Local;
+                        SyntaxReference declaringSyntaxReference = localArgumentDeclaration.DeclaringSyntaxReferences.FirstOrDefault();
+                        if (declaringSyntaxReference is null)
+                        {
+                            return;
+                        }
+                        var variableDeclarator = semanticModel.GetOperationWalkingUpParentChain(declaringSyntaxReference.GetSyntax(cancellationToken), cancellationToken);
+
+                        if (variableDeclarator is IVariableDeclaratorOperation variableDeclaratorOperation)
```

Comment:
```suggestion
                        if (variableDeclaratorOperation != null)
```

Reply:
Can't do this. There'd be warnings/errors about using pattern matching instead